osbrain/tests/test_agent.py

```diff
@@ -91,6 +91,20 @@ def run(self):
     assert a0.ping() == 'pong'
 
 
+def test_agent_configure(nsproxy):
+    class MyAgent(Agent):
+        def configure(self):
+            self.x = 0
+            self.each(.5, 'incr')
+
+        def incr(self):
+            self.x += 1
+
+    agent = run_agent('agent', base=MyAgent)
+    time.sleep(1.2)
+    assert agent.get_attr('x') == 3
```

Comment:
Maybe we need to rethink all this...

I would have said:

```python
assert wait_agent_attr(agent, name='x', value=3, timeout=1.2)
```

But this can break as there is a race condition: we might check the attribute before `.configure()` is executed. Maybe it makes sense to have both methods: `on_init()` for variables/attributes/parameters, and `.configure()` for starting timers, etc..

Not sure about this, will have to think about it.

Reply:
That would make sense. However, although possible, I still don't think we will ever run into that race condition.

Though if you think we still should worry about it, using `on_init()` should be fine.

Reply:
We should always worry about possible race conditions. Sooner or later they bite you and they are a pain in the $%&@# to debug... :wink:

Maybe we could add another attribute `configured` which is set to true after configuration. Maybe we should wait for that variable to be true when we execute `run_agent` rather than waiting for `running` to be true.

What do you think?

Reply:
Yeah, it would be okay. Harder for the user only. But if we do not find a better alternative it might be the best thing to do anyway.

If we go this way we should leave the `.on_init()` in the example and docs just like before and maybe add a paragraph in the Timers section to tell the user about the existence of the `.configure()` method.

Reply:
If you are okay then I would rather use the `on_init()` here for setting `x` and then use the `assert wait_agent_attr()`.